cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -34,6 +34,52 @@ const (
 	statusPackage  = protogen.GoImportPath("google.golang.org/grpc/status")
 )
 
+type serviceGenerateHelper struct {
+	properties map[string]string
+}
+
+func (serviceGenerateHelper) generateProcessedServiceName(service *protogen.Service) string {
+	return fmt.Sprintf("%s", service.Desc.FullName())
+}
+
+func (serviceGenerateHelper) preprocessServiceName(g *protogen.GeneratedFile, service *protogen.Service, clientName string) {
+}
+
+func (serviceGenerateHelper) generateUmplementedHelperCodes(gen *protogen.Plugin, file *protogen.File, g *protogen.GeneratedFile, service *protogen.Service) {
+	serverType := service.GoName + "Server"
+	mustOrShould := "must"
+	if !*requireUnimplemented {
+		mustOrShould = "should"
+	}
+	// Server Unimplemented struct for forward compatibility.
+	g.P("// Unimplemented", serverType, " ", mustOrShould, " be embedded to have forward compatible implementations.")
+	g.P("type Unimplemented", serverType, " struct {")
+	g.P("}")
+	g.P()
+	for _, method := range service.Methods {
+		nilArg := ""
+		if !method.Desc.IsStreamingClient() && !method.Desc.IsStreamingServer() {
+			nilArg = "nil,"
+		}
+		g.P("func (Unimplemented", serverType, ") ", serverSignature(g, method), "{")
+		g.P("return ", nilArg, statusPackage.Ident("Errorf"), "(", codesPackage.Ident("Unimplemented"), `, "method `, method.GoName, ` not implemented")`)
+		g.P("}")
+	}
+	if *requireUnimplemented {
+		g.P("func (Unimplemented", serverType, ") mustEmbedUnimplemented", serverType, "() {}")
+	}
+	g.P()
+}
+
+func (serviceGenerateHelper) generateServerSideHelperCodes(gen *protogen.Plugin, file *protogen.File, g *protogen.GeneratedFile, service *protogen.Service, serverType string, serviceDescVar string) {
+}
+
+type serviceGenerateHelperImpl struct {
```

Comment:
This type seems unnecessary.

Reply:
If `helper` is `serviceGenerateHelper`, then I am not allowed to override it's method internally.
eg: `method generateNewClientDefinitions already declared for type serviceGenerateHelper`

Thus I need a `serviceGenerateHelperImpl` so I can override it's methods internally.

Reply:
Sounds like you want an `interface` instead?

Reply:
Done. Changed it to interface